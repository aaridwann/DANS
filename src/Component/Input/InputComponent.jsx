import React from "react";

function InputComponent({
  value,
  disabled,
  placeholder,
  onChange,
  label,
  type,
  name,
}) {
  return (
    <div className="relative">
      <input
        data-testid="input"
        value={value}
        onChange={onChange}
        autoComplete="off"
        id={name}
        name={name}
        type={type || "text"}
        className="peer bg-white placeholder-transparent h-10 w-full border-cyan-500 border-b-2 text-gray-900 focus:outline-none focus:borer-rose-600"
        placeholder={placeholder}
        disabled={disabled}
      />
      <label
        htmlFor={name}
        className="absolute left-0 -top-3.5 text-gray-600 text-sm peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-440 peer-placeholder-shown:top-2 transition-all peer-focus:-top-3.5 peer-focus:text-gray-600 peer-focus:text-sm"
      >
        {label}
      </label>
    </div>
  );
}

export default React.memo(InputComponent);
