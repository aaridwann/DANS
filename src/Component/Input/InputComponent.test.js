import React from "react";
import { fireEvent, render } from "@testing-library/react";
import InputComponent from "./InputComponent";

describe("Input unit test", () => {
  test("Snapshot testing", () => {
    expect(render(<InputComponent />)).toMatchSnapshot();
  });

  test("render props", () => {
    const Props = {
      name: "username",
      value: "hello world",
      onChange: jest.fn(),
      label: "username",
      type: "text",
      placeholder: "Username",
      disabled: false
    };
    const {
      getByText,
      getByTestId,
      getByLabelText,
      getByPlaceholderText
    } = render(<InputComponent {...Props} />);
    const input = getByTestId("input");
    const placeholder = getByPlaceholderText(Props.placeholder);
    const label = getByLabelText(Props.label);
    fireEvent.change(input, { target: { value: "hello test" } });

    expect(Props.onChange).toBeCalled()
    expect(label).toBeInTheDocument();
    expect(placeholder).toBeInTheDocument();
    expect(input.value).toEqual(Props.value);
    expect(input.type).toEqual(Props.type);
    expect(input.name).toEqual(Props.name);
    expect(input.disabled).toEqual(false);
  });
});
