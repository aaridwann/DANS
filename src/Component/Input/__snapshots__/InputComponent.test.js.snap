// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input unit test Snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <input
          autocomplete="off"
          class="peer bg-white placeholder-transparent h-10 w-full border-b-2 border-gray-300 text-gray-900 focus:outline-none focus:borer-rose-600"
          data-testid="input"
          type="text"
          value=""
        />
        <label
          class="absolute left-0 -top-3.5 text-gray-600 text-sm peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-440 peer-placeholder-shown:top-2 transition-all peer-focus:-top-3.5 peer-focus:text-gray-600 peer-focus:text-sm"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <input
        autocomplete="off"
        class="peer bg-white placeholder-transparent h-10 w-full border-b-2 border-gray-300 text-gray-900 focus:outline-none focus:borer-rose-600"
        data-testid="input"
        type="text"
        value=""
      />
      <label
        class="absolute left-0 -top-3.5 text-gray-600 text-sm peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-440 peer-placeholder-shown:top-2 transition-all peer-focus:-top-3.5 peer-focus:text-gray-600 peer-focus:text-sm"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
